client-app/shared/catalog/components/facet-filter.vue

```diff
@@ -1,43 +1,111 @@
 <template>
-  <VcWidget size="xs" collapsible :title="facet.label" collapsed>
-    <div :class="{ 'fade-bottom': hasFade }">
-      <VcInput
-        v-if="searchFieldVisible"
-        v-model.trim="searchKeyword"
-        size="sm"
-        class="mb-4"
-        maxlength="30"
-        :disabled="loading"
-        :placeholder="$t('common.labels.search', [facet.label])"
-        truncate
-      />
-
-      <div class="-me-1 space-y-3 overflow-y-auto pe-1" :style="{ maxHeight }">
-        <VcCheckbox
-          v-for="item in searchedValues"
-          :key="item.value"
-          v-model="item.selected"
+  <div :class="`facet-filter facet-filter--${mode}`">
+    <!-- Collapsable mode -->
+    <VcWidget v-if="mode === 'collapsable'" size="xs" collapsible :title="facet.label" collapsed>
+      <div :class="{ 'fade-bottom': hasFade }">
+        <VcInput
+          v-if="searchFieldVisible"
+          v-model.trim="searchKeyword"
+          size="sm"
+          class="mb-4"
+          maxlength="30"
           :disabled="loading"
-          @change="changeFacetValues"
+          :placeholder="$t('common.labels.search', [facet.label])"
+          truncate
+        />
+
+        <div class="-me-1 space-y-3 overflow-y-auto pe-1" :style="{ maxHeight }">
+          <VcCheckbox
+            v-for="item in searchedValues"
+            :key="item.value"
+            v-model="item.selected"
+            :disabled="loading"
+            @change="changeFacetValues"
+          >
+            <div :class="['flex text-13', item.selected ? 'font-semibold' : 'font-medium text-gray-500']">
+              <span class="truncate">{{ item.label }}</span>
+              <span class="ml-1">{{ $t("pages.catalog.facet_card.item_count_format", [item.count]) }}</span>
+            </div>
+          </VcCheckbox>
+
+          <div v-if="isNoResults" class="text-sm font-medium">{{ $t("pages.catalog.no_facet_found_message") }}</div>
+
+          <div v-if="isAnchorAdded" ref="fadeVisibilityAnchor" class="!mt-0 h-px"></div>
+        </div>
+      </div>
+
+      <template v-if="isShowMoreVisible" #footer-container>
+        <div class="px-2 py-0.5">
+          <VcButtonSeeMoreLess v-model="isExpanded" />
+        </div>
+      </template>
+    </VcWidget>
+
+    <!-- Dropdown mode -->
+    <VcDropdownMenu v-if="mode === 'dropdown'" :offset-options="4" class="relative z-10" :max-height="maxHeight">
+      <template #trigger="{ opened }">
+        <VcButton
+          class="facet-filter__trigger border-2 border-r-4"
```

Comment:
facet-filter__trigger also used in products-filters.vue. line 112. Lets try to refactor and use one class in one component

Reply:
In `products-filters.vue` the class it's used in `querySelectorAll(".facet-filter__trigger")` to get access for dom elements. So I think it's okay in this case.